src/Xamarin.Android.Build.Tasks/Tasks/ConvertCustomView.cs

```diff
@@ -0,0 +1,152 @@
+// Copyright (C) 2018 Microsoft, Inc. All rights reserved.
+
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Monodroid;
+
+namespace Xamarin.Android.Tasks {
+	public class ConvertCustomView : Task {
+
+		[Required]
+		public string CustomViewMapFile { get; set; }
+
+		[Required]
+		public string AcwMapFile { get; set; }
+
+		public string ResourceNameCaseMap { get; set; }
+
+		public ITaskItem [] ResourceDirectories { get; set; }
+
+		public override bool Execute ()
+		{
+			var resource_name_case_map = MonoAndroidHelper.LoadResourceCaseMap (ResourceNameCaseMap);
+			var acw_map = MonoAndroidHelper.LoadAcwMapFile (AcwMapFile);
+			var customViewMap = MonoAndroidHelper.LoadCustomViewMapFile (BuildEngine4, CustomViewMapFile);
+			var processed = new HashSet<string> ();
+
+			foreach (var kvp in acw_map) {
+				var key = kvp.Key;
+				var value = kvp.Value;
+				if (key == value)
+					continue;
+				if (customViewMap.TryGetValue (key, out HashSet<string> resourceFiles)) {
+					foreach (var file in resourceFiles) {
+						if (processed.Contains (file))
+							continue;
+						if (!File.Exists (file))
+							continue;
+						var document = XDocument.Load (file);
+						var e = document.Root;
+						bool update = false;
+						foreach (var elem in AndroidResource.GetElements (e).Prepend (e)) {
+							update |= TryFixCustomView (elem, acw_map, (t, m) => {
```

Comment:
Is there any way to "unify" this logic with `<ConvertResourceCases/>`?  In particular lines 49-68 here look nearly identical to: https://github.com/xamarin/xamarin-android/blob/87bb898d3e325ca348ca08b32ac0fe4812c9c93a/src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs#L89-L106

Is there any way to unify these instead of copying them?